load_collection.json

```
@@ -268,13 +303,19 @@
                                     }
                                 }
                             }
-                        }
+                        },
+                        "crs": 3857
                     },
                     "result": true
                 }
             }
         }
     ],
+    "exceptions": {
+        "TargetCrsMissing": {
+            "message": "The data requested from the collection has multiple CRS assigned and thus the parameter `crs` must be specified."
```

For Sentinel-2 collections (which on many back-ends have multiple CRS defined), one CRS MUST be specififed even if the default CRS can be a null? I'm confused by this sentence.

---

@lforesta You must specify a non-null target CRS for your Sentinel-2 example.
Maybe the message should be as follows?
```suggestion
            "message": "The data requested from the collection has multiple CRS assigned and thus a valid target CRS must be specified."
```

---

So a CRS can be null only when a unique is present in the collection's metadata and will be used if no other is specified, right? Ok now it's clearer, but I am still not convinced about forcing a unique CRS for the output.

Suppose a group working on algorithms which are entirely scene based (i.e. processing rasters independently); with S2 they might be happy (or even prefer) that the output has the same CRS as the original raster, but with this approach, to achieve their goal they would need to submit one job for each CRS.

---

@lforesta There are no scenes (= granules) in the openEO world, but if a back-end gives users information on scenes and users would want to work on them individually, then I would expect that each scene is an individual data cube. That means crs doesn't need to be set in load_collection (1 scene = 1 crs) and they can work with the original crs until they want to merge and need to resample to a common crs. You can easily use multiple load_collection in a process graph and re-use algorithms with stored process graphs, so it's *not* necessarily one job for each crs. 

The back-end only throws the error if the data requested hast multiple crs assigned. If you load data from a single crs although the collection has multiple crs assigned, you don't need to throw an error. Only if you load data over a region that has multiple crs assigned, you need to make clear what the target crs is. It just needs to be unambiguous.

I think this makes sense, no?

---

Ok never mind the scenes/granules. If my processing is pixel based (i.e. every pixel can be processed completely independently from others), it does not matter that my datacube has multiple CRS or one only. Clearly I'm thinking of a loose datacube, but I think there are people who work with applications of this type. For example, groups doing research on processors from L1 to L2, often they process data on a pixel level. Say I'm one such group and I want to create my S2-L2 product for an entire continent by using an openEO compliant back-end. I could do:
```
load_collection(S2, bbox=full-continent)
apply(my_UDF) # the UDF may be a set of standard processes
save_raster
```

If I'm forced to specify a unique CRS for this example, the back-end will have to reproject lots of data, although the group in this case is maybe happier to have the original UTM tile structure preserved (data like this is then used by other groups to do further processing, they will make sure it's all in the projection they want to use).

In this example, to achieve the same without reprojecting, either I create one job per CRS or within the same job I call load_collection multiple times with different bbox defined, right? This seems a lot of overhead for "keeping the original projection".

Anyway, maybe this is a corner case and in any case there are ways to create jobs where the original multiple CRSs are kept (paragraph above). I just wanted to point out example(s) where this PR may not be optimal.

---

I'm not sure how your example would work with save_result. How would a data cube of a whole continent be stored? When saving, the save_result process doesn't know any longer the original files and thus can't save it to the original tile structure. Maybe some back-ends can do that, but user can't expect that to happen on all back-ends. So this use case doesn't even work without having this PR merged, I think. I don't think your use case can currently be tackled very well in openEO. I guess we would need a process that can load individual tiles into individual data cubes for that, e.g. load_tiles() which returns an array of data cubes and then we'd need an array_each process so you can apply a "callback" on each data cube and save them individually.

---

Hey @lforesta, I think this is an interesting discussion. To not forget about it later, I moved it to #108. Please follow-up there regarding this tile-based use case.